Test failures with Python 3.8

```
=================================== FAILURES ===================================
_________________ TestFancyObjResolver.testWhenSplitOnNextLine _________________

self = <mockito.tests.when2_test.TestFancyObjResolver object at 0x7f45e8ee8ac0>

    def testWhenSplitOnNextLine(self):
        # fmt: off
>       when2(
            os.path.commonprefix, '/Foo').thenReturn(True)

build/lib/mockito/tests/when2_test.py:77: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
build/lib/mockito/mockito.py:243: in when2
    obj, name = get_obj_attr_tuple(fn)
build/lib/mockito/utils.py:159: in get_obj_attr_tuple
    return get_function_host(path)
build/lib/mockito/utils.py:59: in get_function_host
    obj, name = find_invoking_frame_and_try_parse()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def find_invoking_frame_and_try_parse():
        # Actually we just want the first frame in user land; we're open for
        # refactorings here and don't yet decide on which frame exactly we hit
        # that user land.
        stack = inspect.stack(2)[2:10]
        for frame_info in stack:
            # Within `patch` and `spy2` we delegate to `when2` but that's not
            # user land code
            if frame_info[3] in ('patch', 'spy2'):
                continue
    
            source = ''.join(frame_info[4])
            m = FIND_ID.match(source)
            if m:
                # id should be something like `os.path.exists` etc.
                id = m.group(1)
                parts = id.split('.')
                if len(parts) < 2:
                    raise TypeError("can't guess origin of '%s'" % id)
    
                frame = frame_info[0]
                vars = frame.f_globals.copy()
                vars.update(frame.f_locals)
    
                # Now that's a simple reduce; we get the initial value from the
                # locally available `vars`, and then reduce the middle parts via
                # `getattr`. The last path component gets not resolved, but is
                # returned as plain string value.
                obj = vars.get(parts[0])
                for part in parts[1:-1]:
                    obj = getattr(obj, part)
                return obj, parts[-1]
    
>       raise TypeError('could not destructure first argument')
E       TypeError: could not destructure first argument

build/lib/mockito/utils.py:103: TypeError
______________ TestFancyObjResolver.testEnsureWithWhen2SplitLine _______________

self = <mockito.tests.when2_test.TestFancyObjResolver object at 0x7f45e8fc6160>

    def testEnsureWithWhen2SplitLine(self):
        # fmt: off
>       with when2(
                os.path.commonprefix, '/Foo'):

build/lib/mockito/tests/when2_test.py:88: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
build/lib/mockito/mockito.py:243: in when2
    obj, name = get_obj_attr_tuple(fn)
build/lib/mockito/utils.py:159: in get_obj_attr_tuple
    return get_function_host(path)
build/lib/mockito/utils.py:59: in get_function_host
    obj, name = find_invoking_frame_and_try_parse()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

    def find_invoking_frame_and_try_parse():
        # Actually we just want the first frame in user land; we're open for
        # refactorings here and don't yet decide on which frame exactly we hit
        # that user land.
        stack = inspect.stack(2)[2:10]
        for frame_info in stack:
            # Within `patch` and `spy2` we delegate to `when2` but that's not
            # user land code
            if frame_info[3] in ('patch', 'spy2'):
                continue
    
            source = ''.join(frame_info[4])
            m = FIND_ID.match(source)
            if m:
                # id should be something like `os.path.exists` etc.
                id = m.group(1)
                parts = id.split('.')
                if len(parts) < 2:
                    raise TypeError("can't guess origin of '%s'" % id)
    
                frame = frame_info[0]
                vars = frame.f_globals.copy()
                vars.update(frame.f_locals)
    
                # Now that's a simple reduce; we get the initial value from the
                # locally available `vars`, and then reduce the middle parts via
                # `getattr`. The last path component gets not resolved, but is
                # returned as plain string value.
                obj = vars.get(parts[0])
                for part in parts[1:-1]:
                    obj = getattr(obj, part)
                return obj, parts[-1]
    
>       raise TypeError('could not destructure first argument')
E       TypeError: could not destructure first argument

build/lib/mockito/utils.py:103: TypeError
================== 2 failed, 1125 passed, 1 xfailed in 2.29s ===================
```

Looks like the frame info doesn't contain needed info anymore with Python 3.8:
```
FrameInfo(frame=<frame at 0x7f59767aea40, file '/build/python-mockito/src/mockito-python-1.1.1/build/lib/mockito/tests/when2_test.py', line 88, code testEnsureWithWhen2SplitLine>, filename='/build/python-mockito/src/mockito-python-1.1.1/build/lib/mockito/tests/when2_test.py', lineno=88, function='testEnsureWithWhen2SplitLine', code_context=['        # fmt: off\n', '        with when2(\n'], index=1)
```

--- a/.travis.yml
+++ b/.travis.yml
@@ -4,16 +4,13 @@ python:
 - '3.4'
 - '3.5'
 - '3.6'
-matrix:
-  include:
-    - python: 3.7
-      dist: xenial
-      sudo: true
+- '3.7'
+- '3.8'
 install:
-  - if [[ $TRAVIS_PYTHON_VERSION == 3.6 ]]; then pip install flake8; fi
+  - if [[ $TRAVIS_PYTHON_VERSION == 3.8 ]]; then pip install flake8; fi
   - pip install .
 script:
-  - if [[ $TRAVIS_PYTHON_VERSION == 3.6 ]]; then flake8 .; fi
+  - if [[ $TRAVIS_PYTHON_VERSION == 3.8 ]]; then flake8 .; fi
   - py.test
 deploy:
   provider: pypi
@@ -22,7 +19,7 @@ deploy:
     secure: ou5t21x3ysjuRA4oj0oEJIiwffkrsKMoyBL0AhBc+Qq7bxFIEMCdTgfkh1lrWrhGA0xNIAwDHOL9gJrpYaqeLUx6F0mCQc2zRfNzYNf/t4x0+23WsIGQ1HxWGCW9ixLmtXU+zFGK6pUoLZjPdCT0HjZsAjgKOudTv4M1+BlUhFnmAvdmBKjl3jfNY4h5JWbVrhPg6HzMgfNI+vQ7JIFjHZ4a0i2BqEbTMt/2UZGal+Mau0sEO3/y4Ud0LcTRhtA6VA0J7nEcv85q+/JhqmbXTs9h6Bz1KC3V4nMPaaIpGqhrX20eLI6fxULlB/yuBq5jNXSvDMeH9lRyv5AlDUy9NAh++JciXSYYp3p984V/LEwRKM3VyB+ZUUe+KeLN7rk6d/Q2elFW9IHpw9cSsmbl1zrG4GjP+eCpCOw0lrLO6MAijSCGXEzWN+5ViwMDrGCS/6CjRRUBRxcXBebeo6ZB6Wkw+JWdFLW3s/OMzDeVtOEkuP6qdR7VMNn2uYOkPbiDZO4d5UGS09gGMWYasqxP/QJth2yuF95uQmqOhLuGSzI02YS6+L1/Xh2fEmsD8LFF3ATfA0MZ/phHjjvD/ZUmnVgGczW9p1zEohJ9EDQsV4P2fHzNP6nblcx7iBTzKsEsqcjTpOn7UYhFAsyiga17dhcfa5IU2nSb0JzzIeWdM0Q=
   on:
     tags: true
-    python: 3.6
+    python: 3.8
     branch: master
     distributions: sdist bdist_wheel
     repo: kaste/mockito-python

--- a/mockito/utils.py
+++ b/mockito/utils.py
@@ -73,7 +73,7 @@ def find_invoking_frame_and_try_parse():
     # Actually we just want the first frame in user land; we're open for
     # refactorings here and don't yet decide on which frame exactly we hit
     # that user land.
-    stack = inspect.stack(2)[2:10]
+    stack = inspect.stack(3)[2:10]
     for frame_info in stack:
         # Within `patch` and `spy2` we delegate to `when2` but that's not
         # user land code

